### PR TITLE
VMExport: add status Ready, remove token required

### PR DESCRIPTION
**Pull request by jpeimer**
##### Short description:
- Add VirtualMachineExport.Status.READY
- Remove token_secret_ref as the required field (devs changed it to be optional)

### CODE REVIEW

**Comment by jpeimer** [comment]
!verified

File: ocp_resources/virtual_machine_export.py [hunk 1]
```diff
@@ -34,10 +37,8 @@ def __init__(
         yaml_file=None,
         **kwargs,
     ):
-        if not (source_kind and source_name and token_secret_ref) and not yaml_file:
-            raise ValueError(
-                "source_kind, source_name and token_secret_ref or a yaml_file is required"
-            )
+        if not (source_kind and source_name) and not yaml_file:
+            raise ValueError("source_kind and source_name or a yaml_file is required")
```
Comment (myakove) [inline ocp_resources/virtual_machine_export.py:41]:
```
 if not (source_kind and source_name) and not yaml_file:
            raise ValueError("source_kind and source_name or a yaml_file is required")
```

Please move it inside `to_dict` after `if not self.yaml_file:`

**Review by myakove — APPROVED** [review]

**Comment by myakove** [comment]
`Remove token_secret_ref as the required field (devs changed it to be optional)`
Is this changed for 4.12 or 4.13?

**Review by myakove — CHANGES_REQUESTED** [review]
Check my comments

File: ocp_resources/virtual_machine_export.py [hunk 2]
```diff
@@ -14,6 +14,9 @@ class VirtualMachineExport(NamespacedResource):
 
     api_group = NamespacedResource.ApiGroup.EXPORT_KUBEVIRT_IO
 
+    class Status(NamespacedResource.Status):
```
Comment (rnetser) [inline ocp_resources/virtual_machine_export.py:17]: Please put under Resource.Status and use remove from Node.Status

**Review by rnetser — CHANGES_REQUESTED** [review]

**Comment by jpeimer** [comment]
> `Remove token_secret_ref as the required field (devs changed it to be optional)` Is this changed for 4.12 or 4.13?
For both 4.12 and 4.13

**Comment by jpeimer** [comment]
!verified

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Comment by jpeimer** [comment]
!cherry-pick v4.12

**Pull request merged**